FilecoinFoundationWeb/filecoin-foundation
# [UXIT-1539] Events Page · Schedule

## 📝 Description

This pull request introduces a new feature to handle event schedules, including the addition of a schema for schedules, updates to the event data structure, and UI components to display the schedule on event pages. The most important changes include adding a new schedule schema, updating the event data structure, and creating new components for displaying the schedule.

- **Type:** New feature

## Note
For the sake of speed, we have not implemented the mobile dropdown at this stage, instead opting for a sticky horizontal scrollbar. With Filipa unavailable, we don’t yet have a solid solution for handling the x-scrollbar effectively. This is also a pending task for the Allocators table.

The mobile dropdown will also be required for the Security Maturity Model Table of Contents, and at that point, we can refactor both implementations.

## 🛠️ Key Changes

### Schema and Data Structure Updates:
* Added a new `ScheduleSchema` to define the structure of event schedules, including days and events (`src/app/_schemas/event/ScheduleSchema.ts`).
* Updated the `EventFrontMatterSchema` to include the new `schedule` field (`src/app/_schemas/event/FrontMatterSchema.ts`).
* Modified the `convertMarkdownToEventData` function to handle the new `schedule` field (`src/app/_utils/convertMarkdownToEventData.ts`).

### UI Components:
* Created a `ScheduleSection` component to display the schedule section on event pages (`src/app/events/[slug]/components/ScheduleSection.tsx`). ([src/app/events/[slug]/components/ScheduleSection.tsxR1-R17](diffhunk://#diff-12077aeda983adab3b92d95a7156148c8e7a587376d353cf1c9b8bcc7fa4ea4aR1-R17))
* Added a `ScheduleTabs` component to display the schedule in a tabbed format (`src/app/events/[slug]/components/ScheduleTabs.tsx`). ([src/app/events/[slug]/components/ScheduleTabs.tsxR1-R65](diffhunk://#diff-dd2e0f521001f0ac1f13251d226a62dbfeb4fac97a834c5f24a84cf10bb83859R1-R65))

### Event Page Integration:
* Integrated the `ScheduleSection` component into the event page (`src/app/events/[slug]/page.tsx`). ([src/app/events/[slug]/page.tsxR18](diffhunk://#diff-815f182831c560a8fc9dc4acaab794fe338311e80251f17c976d8ec33931e8c1R18), [src/app/events/[slug]/page.tsxR51](diffhunk://#diff-815f182831c560a8fc9dc4acaab794fe338311e80251f17c976d8ec33931e8c1R51), [src/app/events/[slug]/page.tsxR76-R79](diffhunk://#diff-815f182831c560a8fc9dc4acaab794fe338311e80251f17c976d8ec33931e8c1R76-R79))
* Updated the event markdown file to include sample schedule data (`src/content/events/fil-bangkok-2024.md`).

### Utility Functions:
* Added utility functions for formatting dates and times (`src/app/events/[slug]/utils/dateUtils.ts`). ([src/app/events/[slug]/utils/dateUtils.tsR1-R10](diffhunk://#diff-0e33f005e2d49b45dbf3b8595fb7106b1d4164370c030c25894c00716b41a113R1-R10))

### Configuration Update:
* Updated the admin configuration to include the new schedule field (`public/admin/config.yml`).

## 📌 To-Do Before Merging

- [ ] Remove test data

## 🧪 How to Test

- Go to http://localhost:3000/events/fil-bangkok-2024 and verify styling
- Go to `http://localhost:3000/admin/index.html` and try to add a new event with an Event Day schedule.

## 📸 Screenshots

![CleanShot 2024-10-10 at 10 14 15@2x](https://github.com/user-attachments/assets/3e489be9-f944-412e-ba36-8bffbaeeb3c4)
